src/KafkaFlow/Producers/MessageProducer.cs

```diff
@@ -156,19 +152,21 @@ private static void FillContextWithResultMetadata(IMessageContext context, Deliv
 
         private static Message<byte[], byte[]> CreateMessage(IMessageContext context)
         {
-            if (!(context.Message.Value is byte[] value))
+            if (context.Message.Value is not byte[] value)
             {
                 throw new InvalidOperationException(
                     $"The message value must be a byte array to be produced, it is a {context.Message.Value.GetType().FullName}." +
                     "You should serialize or encode your message object using a middleware");
             }
 
-            if (!(context.Message.Key is byte[] key))
+            var key = context.Message.Key switch
             {
-                throw new InvalidOperationException(
-                    $"The message key must be a byte array to be produced, it is a {context.Message.Key.GetType().FullName}." +
-                    "You should serialize or encode your message object using a middleware");
-            }
+                string stringKey => Encoding.UTF8.GetBytes(stringKey),
+                byte[] bytesKey => bytesKey,
+                _ => throw new InvalidOperationException(
```

Comment:
The idea is in the future we create some middleware to encode the partition key, as we have with the serializers. The string conversion was maintained to avoid a silent breaking change.